docs/onboarding/07-query-performance.md

```diff
@@ -0,0 +1,75 @@
+# Chapter 7: Query Performance
+
+The Azure Digital Twins graphs that are constructed by digital twins and relationships can be queried using a custom SQL-like query language, referred to as the Azure Digital Twins query language. You can use this query language to retrieve digital twins according to their properties, models, relationships (or properties of relationships).
+
+- [Query Language reference and common pitfalls](https://docs.microsoft.com/en-us/azure/digital-twins/concepts-query-language)
+- [Query the Azure Digital Twins twin graph](https://docs.microsoft.com/en-us/azure/digital-twins/how-to-query-graph)
+- [How to extract query charges incurred from response](https://docs.microsoft.com/en-us/azure/digital-twins/concepts-query-units)
+
+## Key Points
+
+- In Azure Digital Twins, CPU, IOPS and memory are abstracted away. Instead, the relevant number for measuring both performance and query cost is the Azure Digital Twins Query Unit (QU). The QU is essentially a single unit of generalized computation. Thus, when discussing improving performance in an ADT instance, we care about reducing the QU needed for each query.
+- QU consumption is affected by:
+  - **The complexity of the query.** A query is considered more complex if it has more operations. For instance, up to 5 `JOIN`s are allowed in a single query. Using combination operators (`AND`, `OR`) you can build compound queries of multiple query types (`FROM`, `JOIN`, `WHERE`, etc.)
+  - **The size of the result set.** If you ran two queries of equal complexity, but the second returned 50x the results as the first, the second query would consume more QU.
+- You can learn the QU's consumed by each query by looking at the response header. Each response from ADT has a field called "query-charge", where this measurement is recorded. A developer may find this very handy for metrics.
+- Based on what affects QU, we can deduct that the two best ways to improve query performance are: 1) simplify queries as much as possible, while 2) keeping the results set small.
```

Comment:
Nice job establishing this guideline. This will set some precedence for the team on how we would expect them to develop their queries.